devolving_music/lib/song_scores.py

```diff
@@ -12,165 +13,174 @@
+        return new_dict
+    
+    def get_final_list(self):
 
+        scored_submissions = self.get_scores().copy()
+        
+        # remove all song_submissions with no information
+        info_list = SongScores.get_info_sort(scored_submissions)
+        for sub_info in info_list:
+            i_score = scored_submissions[sub_info].info_score
+            if(i_score == 0):
+                del scored_submissions[sub_info]
+            else:
+                break
+        # sort by postpeakyness
+        peaky_list = SongScores.get_peak_sort(scored_submissions)
+        peak_loc = int(0.7*len(peaky_list))
+        # Break peaky_sorted into two bins pre peak and post peak
+        pre_peak = self.get_dict_from_keys(peaky_list[:peak_loc])
+        post_peak = self.get_dict_from_keys(peaky_list[peak_loc:])
+        # energy_sorted
+        comeup = SongScores.get_energy_sort(pre_peak)
+        cooldown = SongScores.get_energy_sort(post_peak)
+        cooldown = cooldown[::-1]
+
+        # final_list is the song submission keys properly sorted
+        final_list = comeup + cooldown
 
-                       song_submissions: Iterable["SongSubmission"]):
+        return final_list
 
-        scored_submissions = SongSubmission.get_scores(
-            comparison_submissions, song_submissions)
+    def get_quality_list(self,
+            length_limit=300):
 
-        # remove all song_submissions with no information
+        final_quality_list = []
 
-        # sort by peakyness
+        # if len(final_list) is above length limit 
+        # remove_index=check_quality(final_list,len(energy_sorted)-lengthlimit)
+        # final_quality_list=remove(energy_sorted,remove_index)
 
-        # peaky_sorted_submissions=get_peak_sort(scored_submissions)
+        return final_quality_list
+
+    ###
+    # static methods can be moved into helper function
+    ###
+    @staticmethod
+    #sorts in ascending order
+    def get_info_sort(song_dict: "dict[int, ScoreSuite]"):
+        info_submissions = sorted(song_dict,key=lambda sub: song_dict.get(sub).info_score)
+        rightend=0
+        for sub in info_submissions:
+            init=song_dict.get(info_submissions[0]).info_score
+            sub_info=song_dict.get(sub).info_score
+            if sub_info==init:
+                rightend+=1
+            else:
+                break 
+        lowest_info = info_submissions[0:rightend]
+        random.shuffle(lowest_info)
+        other_info = info_submissions[rightend:]
+        info_submissions = lowest_info + other_info
+
+        # return list of keys of dictionary of song objects sorted by info
+        return info_submissions
+
+    @staticmethod
+    #sorts in ascending order
+    def get_energy_sort(song_dict: "dict[int, ScoreSuite]"):
+        energy_submissions = sorted(song_dict,key=lambda sub: song_dict.get(sub).energy_score if song_dict.get(sub).energy_score is not None else -inf)
```

Comment:
Using a list comprehension on a dictionary is highly confusing. I'd at least be sorting `song_dict.keys()`, but using a comprehension or equivalent would be even more clear.